cpp/src/traversal/two_hop_neighbors.cu

```diff
@@ -29,154 +31,96 @@
-                                                               num_edges,
+    grid.x = min((ET) MAXBLOCKS, (num_blocks / 512) + 1);
+    compute_bucket_offsets_kernel<<<grid, block, 0, nullptr>>>(d_exsum_degree,
+                                                               d_block_bucket_offsets,
+                                                               graph.number_of_edges,
                                                                output_size);
-    cudaMemcpy(&block_bucket_offsets[num_blocks], &num_edges, sizeof(IndexType), cudaMemcpyDefault);
+
+    block_bucket_offsets[num_blocks] = graph.number_of_edges;
 
     // Scatter the expanded edge lists into temp space
-    grid.x = min((IndexType) MAXBLOCKS, num_blocks);
-    scatter_expand_kernel<<<grid, block, 0, nullptr>>>(exsum_degree,
-                                                       indices,
-                                                       offsets,
-                                                       block_bucket_offsets,
-                                                       num_verts,
+    grid.x = min((ET) MAXBLOCKS, num_blocks);
+    scatter_expand_kernel<<<grid, block, 0, nullptr>>>(d_exsum_degree,
+                                                       graph.indices,
+                                                       graph.offsets,
+                                                       d_block_bucket_offsets,
+                                                       graph.number_of_vertices,
                                                        output_size,
                                                        num_blocks,
-                                                       first_pair,
-                                                       second_pair);
+                                                       d_first_pair,
+                                                       d_second_pair);
 
+    // TODO:  This would be faster in a hash table (no sorting), unless there's
+    //        some reason that the result has to be sorted
     // Remove duplicates and self pairings
-    auto tuple_start = thrust::make_zip_iterator(thrust::make_tuple(first_pair, second_pair));
+    auto tuple_start = thrust::make_zip_iterator(thrust::make_tuple(d_first_pair, d_second_pair));
     auto tuple_end = tuple_start + output_size;
     thrust::sort(rmm::exec_policy(stream)->on(stream), tuple_start, tuple_end);
     tuple_end = thrust::copy_if(rmm::exec_policy(stream)->on(stream),
                                                             tuple_start,
                                                             tuple_end,
                                                             tuple_start,
-                                                            self_loop_flagger<IndexType>());
+                                                            self_loop_flagger<ET>());
```

Comment:
Should this be `VT` since we are looking at indices pairs and not offsets?
```suggestion
                                                            self_loop_flagger<VT>());
```

Reply:
Fixed